src/FrontendAccountCreation.Web/Controllers/ReprocessorExporter/OrganisationController2.cs

```diff
@@ -114,5 +117,78 @@ void FillSessionWithALoadOfNonsense(ReExTeamMemberRole role)
 
             }
         }
+
+        [HttpGet]
+        [Route(PagePath.TeamMemberDetails)]
+        public async Task<IActionResult> TeamMembersDetails()
+        {
+            var session = await _sessionManager.GetSessionAsync(HttpContext.Session) ?? new OrganisationSession();
+
+            SetBackLink(session, PagePath.TeamMemberRoleInOrganisation);
+            await _sessionManager.SaveSessionAsync(HttpContext.Session, session);
+
+            var teamMember = GetCurrentTeamMember(session);
+
+            if (teamMember != null)
+            {
+                var viewModel = new TeamMemberViewModel
+                {
+                    FullName = teamMember.FullName,
+                    Telephone = teamMember.TelephoneNumber,
+                    Email = teamMember.Email
+                };
+
+                return View(viewModel);
+            }
+
+            return View();
+        }
+
+
+        [HttpPost]
+        [Route(PagePath.TeamMemberDetails)]
+        [AuthorizeForScopes(ScopeKeySection = ConfigKeys.FacadeScope)]
+        public async Task<IActionResult> TeamMembersDetails(TeamMemberViewModel model)
+        {
+            var session = await _sessionManager.GetSessionAsync(HttpContext.Session);
+            if (!ModelState.IsValid)
+            {
+                SetBackLink(session, PagePath.TeamMemberRoleInOrganisation);
+                return View(model);
+            }
+
+            var index = session.CompaniesHouseSession?.CurrentTeamMemberIndex;
+
+            if (index.HasValue && session.CompaniesHouseSession?.TeamMembers != null && index.Value < session.CompaniesHouseSession.TeamMembers.Count)
```

Comment:
Have we done unit testing against this looks like we are trying update the existing record on session won't this overwrite previously added team members? Also I don't see any UT for any methods are you still working on UT?